draft-ietf-quic-tls.md

```diff
@@ -704,86 +740,95 @@ an empty context.  The size of the secret MUST be the size of the hash output
 for the PRF hash function negotiated by TLS.
 
 ~~~
-   client_pp_secret_0 =
-      TLS-Exporter("EXPORTER-QUIC client 1rtt", "", Hash.length)
-   server_pp_secret_0 =
-      TLS-Exporter("EXPORTER-QUIC server 1rtt", "", Hash.length)
+client_pp_secret_0 =
+   TLS-Exporter("EXPORTER-QUIC client 1rtt", "", Hash.length)
+server_pp_secret_0 =
+   TLS-Exporter("EXPORTER-QUIC server 1rtt", "", Hash.length)
 ~~~
 
 These secrets are used to derive the initial client and server packet protection
 keys.
 
-After a key update (see {{key-update}}), these secrets are updated using the
-QHKDF-Expand function.  The QHKDF-Expand function is similar in definition to
-HKDF-Expand-Label defined in Section 7.1 of {{!TLS13}}, but it has a different
-base label and omits the hash argument.  QHKDF-Expand uses the PRF hash function
-negotiated by TLS.  The replacement secret is derived using the existing Secret,
-a Label of "client 1rtt" for the client and "server 1rtt" for the server, and
-the same output Length as the PRF hash function selected by TLS.
-
-~~~
-client_pp_secret_<N+1> =
-  QHKDF-Expand(client_pp_secret_<N>, "client 1rtt", Hash.length)
-server_pp_secret_<N+1> =
-  QHKDF-Expand(server_pp_secret_<N>, "server 1rtt", Hash.length)
-~~~
 
-This allows for a succession of new secrets to be created as needed.
+### Updating 1-RTT Secrets
 
-HKDF-Expand-Label uses HKDF-Expand {{!RFC5869}} as shown:
+After a key update (see {{key-update}}), the 1-RTT secrets are updated using a
+variant of the QHKDF-Expand function.  The QHKDF-Update function uses the same
+basic definition as QHKDF-Expand, but it does not include a connection ID.
 
 ~~~
-    QHKDF-Expand(Secret, Label, Length) =
-         HKDF-Expand(Secret, QuicHkdfLabel, Length)
+QHKDF-Update(Secret, Label, Length) =
+   HKDF-Expand(Secret, QhkdfUpdateInfo, Length)
 ~~~
 
-Where the info parameter, QuicHkdfLabel, is specified as:
+QHKDF-Update always uses the PRF hash function negotiated by TLS.  The `info`
+parameter of HKDF-Expand is encoded as a `QhkdfUpdateInfo` structure as shown:
 
 ~~~
-    struct {
-        uint16 length = Length;
-        opaque label<6..255> = "QUIC " + Label;
-        uint8 hashLength = 0;
-    } QuicHkdfLabel;
+struct {
+   uint16 length = Length;
+   opaque label<6..255> = "QUIC " + Label;
+   uint8 connectionId<0>;
```

Comment:
Is this included just to emphasize that it's zero-length, or does it actually include a byte representing the zero-length array?